### PR TITLE
SV: Add custom case selection

### PR DESCRIPTION
**Pull request by kalletlak**

### CODE REVIEW

File: src/pages/studyView/SummaryHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+    @computed
+    get selectedPatientsCount() {
+        return _.uniq(this.props.selectedSamples.map(sample => sample.uniquePatientKey)).length;
+    }
+
+    @bind
+    private openCases() {
+
+        const firstSample = this.props.selectedSamples[0];
```
Comment (zhx828) [inline src/pages/studyView/SummaryHeader.tsx]: You can't assume `selectedSamples` has an element.

File: src/pages/studyView/SummaryHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        const groupedSamples = _.groupBy(this.props.selectedSamples, sample => sample.studyId);
+        const IncludeStudyId: boolean = Object.keys(groupedSamples).length > 1;
+
+        let navCaseIds = _.map(groupedSamples, (samples) => {
+            return _.map(samples, sample => {
```
Comment (adamabeshouse) [inline src/pages/studyView/SummaryHeader.tsx]: I dont see the point of computing groupedSamples?

File: src/pages/studyView/customCaseSelection/CustomCaseSelection.tsx [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+export default class CustomCaseSelection extends React.Component<ICustomCaseSelectionProps, {}> {
+
+    @observable showCaseIds: boolean = false;
+    @observable caseIdsMode: 'sample' | 'patient' = 'sample';
+    @observable caseIds: string = ''
+    @observable invalidCases: string[] = [];
+    @observable validCases: Sample[] = []
+
+    @computed get sampleSet(): { [id: string]: Sample } {
+        return _.reduce(this.props.selectedSamples, (acc: { [id: string]: Sample }, next) => {
```
Comment (adamabeshouse) [inline src/pages/studyView/customCaseSelection/CustomCaseSelection.tsx]: you can write this as `_.keyBy(this.props.selectedSamples, s=>`${s.studyId}:${s.sampleId}`)`

File: src/pages/studyView/styles.scss [hunk 4]
```diff
@@ -0,0 +1,22 @@
+.studyview-summary-header{
```
Comment (zhx828) [inline src/pages/studyView/styles.scss]: Same here, we don't do dash anymore. All camelcase.

File: src/pages/studyView/customCaseSelection/CustomCaseSelection.tsx (hunk 3 above)
Comment (kalletlak) [reply src/pages/studyView/customCaseSelection/CustomCaseSelection.tsx]: User actually enters `sampleId` or `studyId:sampleId` in text area. 
Is there a login in front-end to convert `studyId:sampleId` to `uniqueSampleKey`?

File: src/pages/studyView/SummaryHeader.tsx (hunk 2 above)
Comment (kalletlak) [reply src/pages/studyView/SummaryHeader.tsx]: Here its used to check whether `studyId` needs to be prefixed or not. and also to order samples by study

File: src/pages/studyView/SummaryHeader.tsx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+}
+
+@observer
+export default class SummaryHeader extends React.Component<ISummaryHeaderProps, {}> {
+
+    @observable private isCustomCaseBoxOpen = false;
+
+    @computed
+    get selectedPatientsCount() {
+        return _.uniq(this.props.selectedSamples.map(sample => sample.uniquePatientKey)).length;
```
Comment (zhx828) [inline src/pages/studyView/SummaryHeader.tsx]: _.uniq is unnecessary.

File: src/pages/studyView/SummaryHeader.tsx (hunk 2 above)
Comment (adamabeshouse) [reply src/pages/studyView/SummaryHeader.tsx]: ah yes, makes sense

File: src/pages/studyView/customCaseSelection/CustomCaseSelection.tsx (hunk 3 above)
Comment (adamabeshouse) [reply src/pages/studyView/customCaseSelection/CustomCaseSelection.tsx]: oh I see, this seems fine and a little different than whats going on in Query

File: src/pages/studyView/SummaryHeader.tsx [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    @bind
+    private openCases() {
+
+        const firstSample = this.props.selectedSamples[0];
+        const groupedSamples = _.groupBy(this.props.selectedSamples, sample => sample.studyId);
+        const IncludeStudyId: boolean = Object.keys(groupedSamples).length > 1;
```
Comment (zhx828) [inline src/pages/studyView/SummaryHeader.tsx]: Shouldn't we only uppercase first letter when it is a component/type/interface?

File: src/pages/studyView/customCaseSelection/CustomCaseSelection.tsx (hunk 3 above)
Comment (zhx828) [reply src/pages/studyView/customCaseSelection/CustomCaseSelection.tsx]: It should be the same though. Just curious how this different from the one in Query

Comment (kalletlak) [reply src/pages/studyView/customCaseSelection/CustomCaseSelection.tsx]: Query page includes some logic if Virtual studies are in selection. eventually this component should be updated to support that functionality and replace it(what's in query page)

Comment (adamabeshouse) [reply src/pages/studyView/customCaseSelection/CustomCaseSelection.tsx]: the difference I was thinking of is that here theres already the selected study, and the samples in the study are already known, so you dont need to fetch more. that may be wrong though and im misunderstanding whats going on here

Comment (kalletlak) [reply src/pages/studyView/customCaseSelection/CustomCaseSelection.tsx]: yes, even that's true.

File: src/pages/studyView/customCaseSelection/styles.module.scss [hunk 7]
```diff
@@ -0,0 +1,55 @@
+.customCaseSelection {
```
[the remark this answers is not in the document]
Comment (kalletlak) [reply src/pages/studyView/customCaseSelection/styles.module.scss]: Its gonna be an independent component right, I think this should be ok.

**Pull request merged**